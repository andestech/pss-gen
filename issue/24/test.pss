/*!
@weakgroup      issue
@{
@file
@author         Luther Lee
@data           2024/12/02
@brief          Test issue 24.


---------------------------------------------------------------------------------------------------

### Test Command
@code{.unparsed}
make
@endcode

@cond
*/
component pss_top {
    action root_a {
        activity {
            do print1;
            do print1_workaround;
        }
    }

    action print1 {
        rand int in [1] val;
        exec body ASM = """print1: {{val}}""";
    }

    action print1_workaround {
        rand int val;
        constraint {
            val in [1];
        }
        exec body ASM = """print1_workaround: {{val}}""";
    }
}

/**@}*/
