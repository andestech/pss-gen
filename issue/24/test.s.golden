print1: 1
print1_workaround: 1
